test/cpp/end2end/async_end2end_test.cc

```diff
@@ -807,7 +807,11 @@ class AsyncEnd2endServerTryCancelTest : public AsyncEnd2endTest {
     EXPECT_FALSE(context->IsCancelled());
     context->TryCancel();
     gpr_log(GPR_INFO, "Server called TryCancel()");
-    EXPECT_TRUE(context->IsCancelled());
+    while (!context->IsCancelled()) {
```

Comment:
Ok, I will read the surrounding code for this more thoroughly to confirm. Hopefully we can merge this as you said.